pkg/engine/apicall/apiCall.go

```diff
@@ -129,16 +144,26 @@ func (a *apiCall) executeServiceCall(ctx context.Context, apiCall *kyvernov1.API
 	}
 	defer resp.Body.Close()
 
+	if !a.config.disableAPICallSizeProtection {
+		if resp.ContentLength < 0 {
+			return nil, fmt.Errorf("content length header must be present.")
+		}
+		if resp.ContentLength > a.config.maxAPICallResponseLength {
+			return nil, fmt.Errorf("content length must be less than max response length of %d.", a.config.maxAPICallResponseLength)
+		}
+	}
+
+	reader := io.LimitReader(resp.Body, a.config.maxAPICallResponseLength)
 	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
-		b, err := io.ReadAll(resp.Body)
+		b, err := io.ReadAll(reader)
```

Comment:
Should we only read the specified size, instead of `ReadAll`?

Reply:
`reader := io.LimitReader(resp.Body, a.config.maxAPICallResponseLength)`

This will limit the size of the reader to reader only `maxAPICallResponseLength` bytes